Homework/homework7/task2.py

```diff
@@ -0,0 +1,23 @@
+"""
+Given two strings.
+
+Return if they are equal when both are typed into
+empty text editors. # means a backspace character.
+Note that after backspacing an empty text, the text will continue empty.
+Examples:
+    Input: s = "ab#c", t = "ad#c"
+    Output: True
+    # Both s and t become "ac".
+    Input: s = "a##c", t = "#a#c"
+    Output: True
+    Explanation: Both s and t become "c".
+    Input: a = "a#c", t = "b"
+    Output: False
+    Explanation: s becomes "c" while t becomes "b".
+"""
+import re
+
+
+def backspace_compare(first: str, second: str) -> bool:
+    backspace = ".?#"
+    return re.sub(backspace, "", first) == re.sub(backspace, "", second)
```

Comment:
Was the 7th lecture about Python or about Perl?

> Some people, when confronted with a problem, think "I know, I'll use regular expressions." Now they have two problems.

So your second problem is to solve it without regular expressions.